src/core/ext/filters/client_channel/http_proxy.c

```diff
@@ -34,12 +34,19 @@
 #include "src/core/lib/support/string.h"
 #include "src/core/lib/slice/b64.h"
 
-static void grpc_get_http_proxy_server(grpc_exec_ctx* exec_ctx,
-                                char **name_to_resolve,
+/**
+ * Parses the 'http_proxy' env var and returns the proxy hostname to resolve or
+ * NULL on error. Also sets 'user_cred' if it is not NULL to user credentials
+ * if present in the 'http_proxy' env var.
+ */
+static char *grpc_get_http_proxy_server(grpc_exec_ctx* exec_ctx,
                                 char **user_cred) {
-  *name_to_resolve = NULL;
+  char *proxy_name = NULL;
+  if(user_cred != NULL) {
```

Comment:
you could simply always do it. If the `if` is true, you set to NULL. Otherwise, it's already NULL, so we don't need the `if`. We can also simplify the wording of the docstring accordingly.